Explain the custom polling

joostlek opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
To relief our eternal pain of explaining.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

joostlek committed:
Update define_custom_polling.md

frenck commented on source/_includes/common-tasks/define_custom_polling.md:
```diff
@@ -1,6 +1,13 @@
+
+The reason why we chose this approach is because this gives you way more flexibility on when to poll.
+
+1. This already works with most of the integrations. Since changing polling interval is something that could be a use case for ever integration, it makes sense to pick an approach that works for every integration without having to add support for every individual integrations.
```
```suggestion
1. This already works with most of the integrations. Since changing the polling interval could be a use case for every integration, it makes sense to pick an approach that works for every integration without having to add support to every individual integration.
```

frenck commented on source/_includes/common-tasks/define_custom_polling.md:
```diff
@@ -1,6 +1,13 @@
+The reason why we chose this approach is because this gives you way more flexibility on when to poll.
+
+1. This already works with most of the integrations. Since changing polling interval is something that could be a use case for ever integration, it makes sense to pick an approach that works for every integration without having to add support for every individual integrations.
+2. You can poll whenever you want. With this approach you can make your polling smarter and more efficient. For example, if you have a rate limited solar panel provider with a max amount of requests per day. It might make sense to lower/stop the polling at night so you can poll more when it actually makes sense.
```
```suggestion
2. You can poll whenever you want. With this approach, you can make your polling smarter and more efficient. For example, if you have a rate-limited solar panel provider with a maximum number of requests per day, it might make sense to lower/stop the polling at night so you can poll more during the day when it matters.
```

joostlek committed:
Update source/_includes/common-tasks/define_custom_polling.md

Co-authored-by: Franck Nijhof <git@frenck.dev>

joostlek committed:
Update source/_includes/common-tasks/define_custom_polling.md

Co-authored-by: Franck Nijhof <git@frenck.dev>

frenck commented:
LGTM 👍

joostlek commented:
Oh and we should decide where this should go, as this is placed on a few integration pages as well

jpbede commented:
Looks also good to me. Thanks a lot 👍

c0ffeeca7 commented on source/_includes/common-tasks/define_custom_polling.md:
```diff
@@ -1,6 +1,13 @@
-If you want to define a specific interval at which your device is being polled for data, you can disable the default polling interval and create your own polling service.
+If you want to define a specific interval at which your device is being polled for data, you can disable the default polling interval and create your own polling automation.
+
+The reason why we chose this approach is because this gives you way more flexibility on when to poll.
```
```suggestion
Creating an automation for polling gives you more flexibility on when to poll.
```
What do you mean by *this approach*? Using an automation?

joostlek replied:
Having users create an automation to customize polling yes

c0ffeeca7 commented on source/_includes/common-tasks/define_custom_polling.md:
```diff
@@ -1,6 +1,13 @@
+
+The reason why we chose this approach is because this gives you way more flexibility on when to poll.
+
+1. This already works with most of the integrations. Since changing the polling interval could be a use case for every integration, it makes sense to pick an approach that works for every integration without having to add support to every individual integration.
```
```suggestion
1. Using the homassistant.update_entity service to update an entity already works with most of the integrations. Since changing the polling interval could be a use case for every integration, it makes sense to pick an approach that works for every integration without having to add support to every individual integration.
```
*This already works*... : What is *this* referring to?

joostlek replied:
Using the `homassistant.update_entity` service to update an entity

c0ffeeca7 committed:
tweaks

c0ffeeca7 committed:
Move intro from reuse topic to common tasks

- the rational does not need to be shown in every integration where the tasks is reused
- slight rephrase of the intro section

c0ffeeca7 committed:
Reduce redundancy

c0ffeeca7 committed:
Tweak

c0ffeeca7 committed:
Rephrase

c0ffeeca7 reviewed (approved):
Thank you, @joostlek :+1:

joostlek reviewed (commented):
No thank you
